packages/compiler-sfc/src/parse.ts

```diff
@@ -311,7 +311,13 @@ function createBlock(
         block.src = p.value && p.value.content
       } else if (type === 'style') {
         if (p.name === 'scoped') {
-          ;(block as SFCStyleBlock).scoped = true
+          if (p.value && p.value.content === 'deep') {
+            ;(block as SFCStyleBlock).scoped = p.value.content
+          } else if (p.value && p.value.content === 'slotted') {
+            ;(block as SFCStyleBlock).scoped = p.value.content
+          } else {
+            ;(block as SFCStyleBlock).scoped = true
+          }
```

Comment:
Why not checking `p.value` and then just do something like

```suggestion
          ;(block as SFCStyleBlock).scoped = p.value.content
```

Maybe with a typecheck if necessary `typeof p.value.content === 'string' ? p.value.content : true`

Reply:
Hi @posva , sorry I did not notice your message sooner. 

Well this was because of TS checks. The type of scoped is forced to be either `boolean`, `deep` or `slotted`. TS does not like checking only against `typeof`.

So it depends whether you want to have forced type of scope to these 3 or to `string|boolean`.

Or I can //@ts-ignore that check or wrap it into variable with type matching scoped from style block, but I tried to match your coding style. 

For not I changed it to this:
```ts
const scoped = p.value && p.value.content
if (scoped === 'deep' || scoped === 'slotted') {
  ;(block as SFCStyleBlock).scoped = scoped  
} else {
  ;(block as SFCStyleBlock).scoped = true
}
```
So which possibility do you think works best?